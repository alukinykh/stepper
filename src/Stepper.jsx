import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Step from './Step'

const WrapStepper = styled.div`
    display: flex;
    flex-dirextion: row;
`
const Button = styled.button`
  background: #ffffff;
  color: #5125da;

  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid #5125da;
  border-radius: 3px;
`;

class Stepper extends PureComponent {
    constructor (props) {
        super(props)
        this.state = {
            currentStep: 0
        }
    }

    handleChangeStep = () => {
        if(this.state.currentStep < this.props.steps.length) {
            this.setState({ currentStep: this.state.currentStep + 1 })
        }
    }

  render() {
    const { steps } = this.props
    return (
      <WrapStepper>
        {steps.slice(0, 5).map((step, index) => (
          <Step
            {...step}
            key={step.id} 
            complete={index < this.state.currentStep ? true : false}
            isFirst={index === 0} 
            currentStep={this.state.currentStep}
        />
        ))}
        <Button onClick={this.handleChangeStep}>Next</Button>
      </WrapStepper>
    )
  }
}

Stepper.propTypes = {
  steps: PropTypes.arrayOf(PropTypes.object),
}

Stepper.defaultProps = {
  steps: [{
    id: 0,
    label: 'Step 0',
  },
  {
    id: 1,
    label: 'Step 1',
  }],
}

export default Stepper
