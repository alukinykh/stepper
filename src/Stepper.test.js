import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import Stepper from './Stepper'

const mockSteps = [
  { id: 699, label: 'First Item' },
  { id: 6879, label: 'Seconf Item' },
  { id: 2, label: 'Third Item' },
  { id: 89, label: 'Forth Item' },
]

it('renders component', () => {
  const component = renderer.create(<Stepper steps={mockSteps} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


describe('Stepper component', () => {
  let stepper

  beforeEach(() => {
    jest.resetAllMocks()
    stepper = mount(<Stepper steps={mockSteps} />)
  })

  test('change current step', () => {
    expect(stepper.state().currentStep).toBe(0)
    const button = stepper.find('button')
    button.props().onClick()
    expect(stepper.state().currentStep).toBe(1)
    button.props().onClick()
    expect(stepper.state().currentStep).toBe(2)
  })

  test('check props of first step', () => {
    const firstStep = stepper.find('Step').first().props()
    expect(firstStep.id).toEqual(mockSteps[0].id)
    expect(firstStep.label).toEqual(mockSteps[0].label)
    expect(firstStep.complete).toEqual(false)
    expect(firstStep.isFirst).toEqual(true)
  })
})
