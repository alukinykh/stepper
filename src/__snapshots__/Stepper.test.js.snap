// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component 1`] = `
<div
  className="sc-EHOje dbaRmk"
>
  <div
    className="sc-bdVaJa iwRrDF"
  >
    <div
      className="sc-bwzfXH hZWFyw"
    >
      <div
        className="sc-htpNat hymTUh"
        size={20}
      />
      <div
        className="sc-bxivhb conwii"
      >
        First Item
      </div>
    </div>
  </div>
  <div
    className="sc-bdVaJa iwRrDF"
  >
    <div
      className="sc-ifAKCX khxicc"
    />
    <div
      className="sc-bwzfXH hZWFyw"
    >
      <div
        className="sc-htpNat hymTUh"
        size={20}
      />
      <div
        className="sc-bxivhb conwii"
      >
        Seconf Item
      </div>
    </div>
  </div>
  <div
    className="sc-bdVaJa iwRrDF"
  >
    <div
      className="sc-ifAKCX khxicc"
    />
    <div
      className="sc-bwzfXH hZWFyw"
    >
      <div
        className="sc-htpNat hymTUh"
        size={20}
      />
      <div
        className="sc-bxivhb conwii"
      >
        Third Item
      </div>
    </div>
  </div>
  <div
    className="sc-bdVaJa iwRrDF"
  >
    <div
      className="sc-ifAKCX khxicc"
    />
    <div
      className="sc-bwzfXH hZWFyw"
    >
      <div
        className="sc-htpNat hymTUh"
        size={20}
      />
      <div
        className="sc-bxivhb conwii"
      >
        Forth Item
      </div>
    </div>
  </div>
  <button
    className="sc-bZQynM ccclwR"
    onClick={[Function]}
  >
    Next
  </button>
</div>
`;
